### PR TITLE
#8969 Update cesium to version 1.106.1

### PR DESCRIPTION
**Pull request by allyoucanmap**
## Description

This PR includes the update of Cesium to the latest release 1.106.1

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Minor changes to existing features

## Issue

**What is the current behavior?**

#8969

**What is the new behavior?**

Cesium updated to version 1.106.1

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [x] No

### CODE REVIEW

File: web/client/utils/styleparser/__tests__/CesiumStyleParser-test.js [hunk 1]
```diff
@@ -592,59 +592,6 @@ describe('CesiumStyleParser', () => {
             }).catch(done);
         });
 
-        it('should add leader line where HeightReference is Relative and sampleTerrain is using when from Cesium', (done) => {
```
Comment (offtherailz) [inline web/client/utils/styleparser/__tests__/CesiumStyleParser-test.js:595]: This test is not needed anymore?

**Review by offtherailz — CHANGES_REQUESTED** [review]
- I ' m not allowed to see photo-realistic 3D 
http://localhost:8081/#/viewer/44034
But I think it is because I'm not executing it on dev. Correct?
I tested a couple of things and I got these inconsistencies.

- [ ] BIL terrain do not work anymore. Here a sample map testing it
http://localhost:8081/#/context/test-terrain/44826
Error: 
image

It works on dev

https://dev-mapstore.geosolutionsgroup.com/mapstore/#/context/test-terrain/44826

The same for http://localhost:8081/#/context/Terrain_DTM25_lowi_3857_lzw , http://localhost:8081/#/context/Genova3DContext and http://localhost:8081/#/context/austrocontrol-terrain

File: web/client/utils/styleparser/__tests__/CesiumStyleParser-test.js (hunk 1 above)
Comment (allyoucanmap) [reply web/client/utils/styleparser/__tests__/CesiumStyleParser-test.js:595]: No, it is not needed. They removed the `Cesium.when` library so this case does not exist anymore

**Comment by allyoucanmap** [comment]
> I ' m not allowed to see photo-realistic 3D
> http://localhost:8081/#/viewer/44034
> But I think it is because I'm not executing it on dev. Correct?

  @offtherailz yes, correct

**Review by offtherailz — APPROVED** [review]
This doesn't work on both dev and locally
http://localhost:8081/#/context/Terrain_DTM25_lowi_3857_lzw 

https://dev-mapstore.geosolutionsgroup.com/mapstore/#/context/Terrain_DTM25_lowi_3857_lzw 

(tried to switch to 3d and zoom to the layer that should use the DTM). 

Instead the others work. So I think it is fine

**Pull request merged**